src/modules/content/licenses.js

```diff
@@ -43,14 +43,40 @@ export function addLicenseCommentToHtmlTop(codeSample) {
   }
 }
 
+export function makeLicenseUrl(owner, repo) {
+  return `${githubApiDomain}/repos/${owner}/${repo}/license`;
+}
+
+// https://docs.github.com/en/rest/reference/licenses
+export async function fetchGithubLicense(owner, repo) {
+  const res = await fetch(makeLicenseUrl(owner, repo));
+
+  // TODO:  error response with 403 would usually be because of API rate limit
+  // "message":"API rate limit exceeded for
+  // next in line would be to make a readable notification for that
+
+  if (res.status === 404) {
+    throw new Error('license file not found');
+  }
+
+  return await res.json();
+}
+
+export async function fetchGithubLicenseDetails(licenseKey) {
+  const res = await fetch(`${githubApiDomain}/licenses/${licenseKey}`);
+  return await res.json();
+}
+
 async function processLicenseData(owner, repo, url) {
   let license;
+  let licenseOriginUrl;
   let licenseInfo;
   let licenseAllowed = false;
 
   try {
     const githubLicenseData = await fetchGithubLicense(owner, repo);
     license = githubLicenseData.license;
+    licenseOriginUrl = githubLicenseData.html_url;
```

Comment:
extra thing -- get and pass the origin license url to provide the reference for the author's license.